lib/src/main/java/graphql/nadel/validation/NadelHydrationWhenConditionValidation.kt

```diff
@@ -26,17 +25,19 @@ internal class NadelHydrationWhenConditionValidation() {
             (parent.overall as GraphQLFieldsContainer).getField(whenConditionSourceFieldName)
                 ?: return NadelSchemaValidationError.WhenConditionSourceFieldDoesNotExist(
                     whenConditionSourceFieldName,
-                    overallField
+                    overallField,
                 )
 
-        if (whenConditionSourceField.type.unwrapNonNull() !is GraphQLScalarType) {
+        val whenConditionSourceFieldType = whenConditionSourceField.type.unwrapNonNull()
+        if (whenConditionSourceFieldType !is GraphQLScalarType) {
             return NadelSchemaValidationError.WhenConditionUnsupportedFieldType(
                 whenConditionSourceFieldName,
                 GraphQLTypeUtil.simplePrint(whenConditionSourceField.type),
-                overallField
+                overallField,
             )
         }
-        val whenConditionSourceFieldTypeName: String = (whenConditionSourceField.type as GraphQLScalarType).name
```

Comment:
Avoid manual casting. There is a bug where a non-nullable type causes this to crash.

Reply:
Thanks for fixing this 👍